### PR TITLE
Add chrome$webRequest

### PR DESCRIPTION
**Pull request by briancollins**
This adds type definitions for the `webRequest` API. I went through the documentation manually, so there could be some mistakes, but it is working correctly for the project that I am working on.

This will help with https://github.com/Shraymonks/flow-interfaces-chrome/issues/1

### CODE REVIEW

**Comment by briancollins** [comment]
Thanks for the great library, by the way!

File: interfaces/webRequest.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+type chrome$OnResponseStartedOptions = 'responseHeaders';
+type chrome$OnBeforeRedirectOptions = 'responseHeaders';
+type chrome$OnCompletedOptions = 'responseHeaders';
+type chrome$RequestFilter = {
+  urls: Array<string>,
+  types?: Array<chrome$ResourceType>,
+  tabId?: number,
+  windowId?: number,
+};
+type chrome$HttpHeaders = Array<{name: string, value?: any, binaryValue?: any}>;
```
Comment (Shraymonks) [inline interfaces/webRequest.js]: Based on https://developer.chrome.com/extensions/webRequest#type-HttpHeaders

Sounds like this should be something like:
```
Array<{name: string, value: any} | {name:string, binaryValue: any}>
```

File: interfaces/webRequest.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+type chrome$BlockingResponse = {
+  cancel?: boolean,
+  redirectUrl?: string,
+  requestHeaders?: chrome$HttpHeaders,
+  responseHeaders?: chrome$HttpHeaders,
+  authCredentials?: {username: string, password: string},
+};
+type chrome$UploadData = {
+  bytes?: any,
```
Comment (Shraymonks) [inline interfaces/webRequest.js]: `ArrayBuffer` instead of `any`?

File: interfaces/webRequest.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  file?: string,
+};
+type chrome$FormDataItem = any;
```
Comment (Shraymonks) [inline interfaces/webRequest.js]: `ArrayBuffer | string`?

File: package.json [hunk 4]
```diff
@@ -3,5 +3,5 @@
   "description": "Flow interface declarations for the Chrome Extensions API",
   "name": "flow-interfaces-chrome",
   "repository": "shraymonks/flow-interfaces-chrome",
-  "version": "0.4.0"
+  "version": "0.5.0"
```
Comment (Shraymonks) [inline package.json]: Remove this, I'll bump the version afterwards.

**Review by Shraymonks — CHANGES_REQUESTED** [review]
Thanks for adding this!

**Comment by briancollins** [comment]
Thanks for the review!

**Review by Shraymonks — APPROVED** [review]

**Comment by ssorallen** [comment]
This introduced an error in v0.5.0:

    Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/flow-interfaces-chrome/interfaces/webRequest.js:241:29

    Cannot resolve name chrome$OnErrorOccurredOptions.

        238│         error: string,
        239│       }) => ?chrome$BlockingResponse,
        240│       filter?: chrome$RequestFilter,
        241│       extraInfoSpec?: Array<chrome$OnErrorOccurredOptions>,
        242│     ): void,
        243│   },
        244│ };

I will open an issue and send a PR.

**Pull request merged**